test/testastutils.cpp

```diff
@@ -256,6 +265,58 @@ class TestAstUtils : public TestFixture {
                "}\n";
         ASSERT_EQUALS(false, isVariableChangedByFunctionCall(code, "x ) ;", &inconclusive));
         TODO_ASSERT_EQUALS(false, true, inconclusive);
+
+        code = "void g(int* p);\n"
+               "void f(int x) {\n"
+               "    return g(&x);\n"
+               "}\n";
+        ASSERT_EQUALS(true, isVariableChangedByFunctionCall(code, "x ) ;", &inconclusive));
+        ASSERT_EQUALS(true, inconclusive);
+
+        code = "void g(const int* p);\n"
+               "void f(int x) {\n"
+               "    return g(&x);\n"
+               "}\n";
+        ASSERT_EQUALS(false, isVariableChangedByFunctionCall(code, "x ) ;", &inconclusive));
+        ASSERT_EQUALS(true, inconclusive);
+
+        code = "void g(int** pp);\n"
+               "void f(int* p) {\n"
+               "    return g(&p);\n"
+               "}\n";
+        ASSERT_EQUALS(true, isVariableChangedByFunctionCall(code, "p ) ;", &inconclusive));
+        ASSERT_EQUALS(true, inconclusive);
+
+        code = "void g(int* const* pp);\n"
+               "void f(int* p) {\n"
+               "    return g(&p);\n"
+               "}\n";
+        ASSERT_EQUALS(false, isVariableChangedByFunctionCall(code, "p ) ;", &inconclusive));
```

Comment:
This should be true because the `indirect` will be set to 1, which will check that the data is modified and not the pointer.

Reply:
If `indirect` is set to 0 then it should be `false`.

Reply:
With the new formula and `indirect` always 0 (as it was before), this fails:
~~~
        code = "void g(int a[2]);\n"
               "void f() {\n"
               "    int b[2] = {};\n"
               "    return g(b);\n"
               "}\n";
        ASSERT_EQUALS(true, isVariableChangedByFunctionCall(code, "b ) ;", &inconclusive));
        ASSERT_EQUALS(true, inconclusive);
~~~

Reply:
We need to check for array and up the indirection since arrays are passed by pointer and not by value.

Reply:
Well, a `std::array` can be passed by value, and since we like to conflate those...

Reply:
`inconclusive` was not reset (it is only ever set to `true`, never `false`. Some results don't seem consistent either.

Reply:
That's one of the things I want to address with the work started in #4798.